tests/sync/app.cpp

```diff
@@ -1688,14 +1745,10 @@ TEST_CASE("app: link_user", "[sync][app]") {
                       [&](std::shared_ptr<SyncUser> user, Optional<app::AppError> error) {
             CHECK(!error);
             REQUIRE(user);
-            CHECK(user->identity() != sync_user->identity());
-            CHECK(sync_user->provider_type() == IdentityProviderUsernamePassword);
-            CHECK(user->provider_type() == IdentityProviderFacebook);
+            CHECK(user->identity() == sync_user->identity());
```

Comment:
Maybe check that there are now 2 entries in the identities array indicating that linking succeded?